src/sentry/tasks/statistical_detectors.py

```diff
@@ -241,6 +260,31 @@ def _detect_transaction_trends(
     )
 
 
+def query_transactions_timeseries(
+    transactions: List[Tuple[int, int | str]],
+    start: datetime,
+    agg_function: str,
+) -> Generator[Tuple[int, int, Any], None, None]:
+    end = start.replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+    interval = 3600  # 1 hour
+
+    for project_id, transaction_name in transactions:
+        params: Dict[str, Any] = {
+            "start": end - timedelta(days=14),
+            "end": end,
+            "project_id": [project_id],
+            "project_objects": [Project.objects.get(id=project_id)],
+        }
+        results = metrics_performance.timeseries_query(
```

Comment:
Ideally this would probably use `metrics_performance.bulk_timeseries_query`, but this should be fine for the internal release.

Reply:
I'd start by batching multiple transactions in a single query first. The reason being we want to minimize the number of concurrent queries we make to clickhouse. So where possible, try to query multiple timeseries in a single query.

Reply:
I think we should still try minimizing the number of queries we're making here. You can maybe batch multiple transaction timeseries in a single query with a `Column("transaction")` groupBy.

Snuba allows 10,000 data points per request. 14 days * 1hr * 24hr = 336 data points per transaction name. We can safely get 25 transaction name timeseries